rotkehlchen/chain/ethereum/names.py

```diff
@@ -65,58 +64,146 @@ def find_ens_mappings(
 def search_for_addresses_names(
         database: DBHandler,
         addresses: List[ChecksumEvmAddress],
-) -> Dict[ChecksumEvmAddress, str]:
+) -> Dict[ChecksumEvmAddress, Optional[str]]:
     """This method searches for all names of provided addresses known to rotki. We can show
-    only one name per address, and thus we prioritize known names. Priority is the following:
-    blockchain account labels -> private addressbook -> global addressbook
-    -> ethereum tokens -> hardcoded mappings -> ENS names.
+    only one name per address, and thus we prioritize known names. Priority is read from settings.
     """
-    db_addressbook = DBAddressbook(database)
-    db_ens = DBEns(database)
+
+    prioritizer = NamePrioritizer(database)
+    prioritizer.add_fetchers({
+        'blockchain_account': blockchain_address_to_name,
+        'global_addressbook': global_addressbook_address_to_name,
+        'private_addressbook': private_addressbook_address_to_name,
+        'ethereum_tokens': token_mappings_address_to_name,
+        'hardcoded_mappings': hardcoded_address_to_name,
+        'ens_names': ens_address_to_name,
+    })
 
     with database.conn.read_ctx() as cursor:
-        reverse_ens = db_ens.get_reverse_ens(cursor, addresses=addresses).values()
-        mapped_tokens = GlobalDBHandler().get_tokens_mappings(addresses=addresses)
-        hardcoded_mappings = {}
+        settings = database.get_settings(cursor)
+
+    prioritized_addresses = prioritizer.get_prioritized_names(
+        settings.address_name_priority,
+        addresses,
+    )
+
+    return prioritized_addresses
+
+
+FetcherFunc = Callable[[DBHandler, ChecksumEvmAddress], Optional[str]]
+
+
+class NamePrioritizer:
+    _fetchers: Dict[AddressNameSource, FetcherFunc] = {}
+    _db: DBHandler
+
+    def __init__(self, database: DBHandler):
+        self._db = database
```

Comment:
Don't put attributes at the class level. These are going to be for every instance of name prioritize which is definitely not what you want. They should be in the `init()`

```suggestion

    def __init__(self, database: DBHandler):
        self._fetchers: Dict[AddressNameSource, FetcherFunc] = {}
        self._db = database
```